presto-main/src/main/java/com/facebook/presto/cost/UnnestStatsRule.java

```diff
@@ -0,0 +1,75 @@
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.cost.ComposableStatsCalculator.Rule;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.UnnestNode;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.planner.plan.Patterns.unnest;
+import static java.lang.Double.NaN;
+
+public class UnnestStatsRule
+        implements Rule<UnnestNode>
+{
+    private static final int UPPER_BOUND_ROW_COUNT_FOR_ESTIMATION = 1;
+    @Override
+    public Pattern<UnnestNode> getPattern()
+    {
+        return unnest();
+    }
+
+    @Override
+    public Optional<PlanNodeStatsEstimate> calculate(UnnestNode node, StatsProvider statsProvider, Lookup lookup, Session session, TypeProvider types)
+    {
+        PlanNodeStatsEstimate sourceStats = statsProvider.getStats(node.getSource());
+        // Since we don't have stats for cardinality about the unnest column, we cannot estimate the row count.
+        // However, when the source row count is low, the error would not matter much in query optimization.
+        // Thus we'd still populate the inaccurate numbers just so stats are populated to enable optimization
+        // potential.
+        PlanNodeStatsEstimate.Builder calculatedStats = PlanNodeStatsEstimate.builder();
+        if (sourceStats.getOutputRowCount() <= UPPER_BOUND_ROW_COUNT_FOR_ESTIMATION) {
```

Comment:
I would make this rule applicate only if the input is a single row. If the input is multi-row - i would return an unknown estimate.

Reply:
Do you feel strong about leaving it enabled (providing column stats) when this check doesn't pass? I would simply disable it (return empty).

Reply:
For the purpose that makes me add this, no. So happy to just return empty if there are more rows.